src/bscPlugin/codeActions/XmlFileCodeActionsProcessor.ts

```diff
@@ -0,0 +1,44 @@
+import type { CodeAction, Range } from 'vscode-languageserver';
+import { CodeActionKind } from 'vscode-languageserver';
+import { DiagnosticCodeMap } from '../../DiagnosticMessages';
+import type { XmlFile } from '../../files/XmlFile';
+import type { BsDiagnostic } from '../../interfaces';
+import util from '../../util';
+
+export class XmlFileCodeActionsProcessor {
```

Comment:
You really do like classes ;)

Reply:
Yep! :D Methods on direct module exports are much more annoying to mock/stub in unit tests. Also, for this specific set of classes, it vastly simplifies the individual method parameters because the constructor receives and maintains the state.

Reply:
Please: more classes!! It's hard enough to anything in a performant way as it is. There's a massive learning curve in getting up to speed as it is. More ooad will help mitigate that and get more contributions.